### PR TITLE
perlPackages.PDL and dep

### PR DESCRIPTION
**Pull request by invokesus**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Please merge/build after https://github.com/NixOS/nixpkgs/pull/84600 is merged. Thanks!

### CODE REVIEW

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -14747,6 +14762,52 @@ let
...
+    nativeBuildInputs = with pkgs; [ autoPatchelfHook libGL.dev glibc.dev mesa_glu.dev ];
+
+    buildInputs = [ DevelChecklib TestDeep TestException TestWarn ] ++
+                  (with pkgs; [ gsl freeglut xorg.libXmu xorg.libXi ]);
+
+    propagatedBuildInputs = [
+      AstroFITSHeader
+      ConvertUU
+      ExtUtilsF77
+      FileMap
```
Comment (cole-h) [inline pkgs/top-level/perl-packages.nix:14796]:
```
undefined variable 'FileMap' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/packet-spot-eval-1/pkgs/top-level/perl-packages.nix:14838:7
```

Comment (invokesus) [reply pkgs/top-level/perl-packages.nix:14796]: As I mention above, this needs #84600 to be merged first. :)

Comment (cole-h) [reply pkgs/top-level/perl-packages.nix:14796]: Hehe, I should pay closer attention. Sorry!

Comment (invokesus) [reply pkgs/top-level/perl-packages.nix:14796]: np. Thanks for reviewing! 🙂

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -14747,6 +14762,52 @@ let
...
+        --replace 'POSIX_THREADS_INC  => undef' 'POSIX_THREADS_INC  => "-I${pkgs.glibc.dev}/include"' \
+        --replace 'WITH_MINUIT => undef' 'WITH_MINUIT => 0' \
+        --replace 'WITH_SLATEC => undef' 'WITH_SLATEC => 0' \
+        --replace 'WITH_HDF => undef' 'WITH_HDF => 0' \
+        --replace 'WITH_GD => undef' 'WITH_GD => 0' \
+        --replace 'WITH_PROJ => undef' 'WITH_PROJ => 0'
+
+      # This unit test attempts to write to HOME(/homeless-shelter) with "~".
+      # Patch it to write to /tmp instead.
+      substituteInPlace t/fits.t --replace '~' '/tmp'
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:14784]: Attempted to build this PR (with https://github.com/NixOS/nixpkgs/pull/84600 merged), and one of the tests fail. Perhaps due to substituteInPlace unintentionally replacing ~ in some pattern matching operators.

Latest upstream version might not require this replace.

Relevant parts of the output:
```
$ nix-build -A perlPackages.PDL
[...]
t/fits.t ........................ 1/97 Unknown regexp modifier "/b" at t/fits.t line 324, at end of line
Unmatched ) in regex; marked by <-- HERE in m/tmp) <-- HERE / at t/fits.t line 324.
# Looks like your test exited with 255 just after 1.
t/fits.t ........................ Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 96/97 subtests
[...]
Test Summary Report
-------------------
[...]
t/fits.t                      (Wstat: 65280 Tests: 1 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 97 tests but ran 1.
[...]
Files=135, Tests=1536, 49 wallclock secs ( 0.59 usr  0.15 sys + 31.08 cusr 17.85 csys = 49.67 CPU)
Result: FAIL
Failed 1/135 test programs. 0/1536 subtests failed.
make: *** [Makefile:1103: test_dynamic] Error 255
builder for '/nix/store/4jj8ii0c8cw5lq5axap78c9gfn3hw5ml-perl5.30.1-PDL-2.019.drv' failed with exit code 2
e
```

File: pkgs/top-level/perl-packages.nix [hunk 3]
```diff
@@ -14762,6 +14762,52 @@ let
     };
   };
 
+  PDL = buildPerlPackage rec {
+    pname = "PDL";
+    version = "2.019";
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:14767]:
```suggestion
    version = "2.021";
```

File: pkgs/top-level/perl-packages.nix [hunk 4]
```diff
@@ -14747,6 +14762,52 @@ let
...
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/C/CH/CHM/${pname}-${version}.tar.gz";
+      sha256 = "51a47beedb1dd5bb4b6f6441a0ad38d4c95d7cbf3bf6b1b4c9128732ea2d77fe";
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:14770]: Latest PDL is maintained by EDJ https://metacpan.org/pod/PDL

```suggestion
      url = "mirror://cpan/authors/id/E/ET/ETJ/${pname}-${version}.tar.gz";
      sha256 = "1szgyivnjiqrlclvh14ccg3vyfzlnz59bjqchlcrjbkirkd3y331";
```

**Comment by invokesus** [comment]
Amended and force-pushed. 🙂

**Comment by stigtsp** [comment]
@GrahamcOfBorg perlPackages.PDL perlPackages.AstroFITSHeader perldevelPackages.PDL perldevelPackages.AstroFITSHeader

**Comment by stigtsp** [comment]
@invokesus pls change the commit message `perlPackages.Astro-FITS-Header: init at 3.07` to `perlPackages.AstroFITSHeader: init at 3.07`

**Comment by stigtsp** [comment]
@GrahamcOfBorg build perlPackages.PDL perlPackages.AstroFITSHeader perldevelPackages.PDL perldevelPackages.AstroFITSHeader

**Comment by invokesus** [comment]
> @invokesus pls change the commit message `perlPackages.Astro-FITS-Header: init at 3.07` to `perlPackages.AstroFITSHeader: init at 3.07`

Done. :+1:

File: pkgs/top-level/perl-packages.nix [hunk 5]
```diff
@@ -804,6 +804,21 @@ let
...
   };
 
+  AstroFITSHeader = buildPerlModule rec {
+    pname = "Astro-FITS-Header";
+    version = "3.07";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/T/TJ/TJENNESS/${pname}-${version}.tar.gz";
+      sha256 = "530d59ef0c0935f9862d187187a2d7583b12c639bb67db14f983322b161892d9";
+    };
+    buildInputs = [ ModuleBuild ];
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:814]: `buildPerlModule` already has `ModuleBuild`

```suggestion
```

Need to wait to merge util https://github.com/NixOS/nixpkgs/pull/97098/commits/340f23f171b13875263881b637d6924dcc5045d7 hits `master`

**Review by stigtsp — CHANGES_REQUESTED** [review]
This package is not supported on `darwin` since it depends on glibc. Consider setting `platforms = platforms.linux`

**Comment by stigtsp** [comment]
Tested `pdl` executable OK. Ran `demo 3d` without problems.

**Comment by invokesus** [comment]
Thanks again for the quick review. Really appreciate it!

**Comment by invokesus** [comment]
Fixed my typo and added the suggested changes. 🙂

**Review by stigtsp — APPROVED** [review]
LGTM

**Comment by stigtsp** [comment]
@GrahamcOfBorg build perlPackages.PDL perldevelPackages.PDL

**Pull request merged**